apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/ReadTests.cs

```diff
@@ -246,26 +248,67 @@ public class ReadTests : IClassFixture<TestDatabaseFixture> {
         [Fact]
         public async Task TestSelectOne()
         {
-            Exception exceptionCaught = null;
-            long result = 0;
-            try
-            {
-                using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
-                {
-                    var cmd =
-                        connection.CreateSelectCommand("SELECT 1");
-                    result = await cmd.ExecuteScalarAsync<long>();
-                }
-            }
-            catch (Exception e)
-            {
-                exceptionCaught = e;
-            }
-
-            Assert.Null(exceptionCaught);
+            long result = await ExecuteAsync<long>("SELECT 1");
             Assert.Equal(1, result);
         }
 
+        [Fact]
+        public async Task TestPositiveInf()
+        {
+            double result = await ExecuteAsync<double>("SELECT IEEE_DIVIDE(1, 0)");
+            Assert.True(double.IsPositiveInfinity(result));
+        }
+
+        [Fact]
+        public async Task TestNegativeInf()
+        {
+            double result = await ExecuteAsync<double>("SELECT IEEE_DIVIDE(-1, 0)");
+            Assert.True(double.IsNegativeInfinity(result));
+        }
+
+        [Fact]
+        public async Task TestNaN()
```

Comment:
hmm not exactly sure what you mean.  do you mean different binary representations for NaN?